### ISSUE TITLE
[Bug]: Error on generation!

### ISSUE THREAD

**Issue by navin-hariharan** [continue-revolution/sd-forge-animatediff#4]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### Have you read FAQ on README?

- [X] I have updated WebUI and this extension to the latest version

### What happened?

RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
*** Error completing request

Traceback (most recent call last):
      File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\call_queue.py", line 57, in f
        res = list(func(*args, **kwargs))
    TypeError: 'NoneType' object is not iterable

### Steps to reproduce the problem

1. Start the generation with animatediff!

### What should have happened?

A gif should have been generated

### Commit where the problem happens

webui: d81e353d8928147bbd973068d0efbb2802affe0f
extension: https://github.com/continue-revolution/sd-forge-animatediff/commit/c25a46d3dfacb60ea0d485d71f1aa78db2788d35

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
No
```

### Console logs

```Shell
Moving model(s) has taken 3.57 seconds
  0%|                                                                                                                                                                                      | 0/30 [00:06<?, ?it/s]
Traceback (most recent call last):
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules_forge\main_thread.py", line 37, in loop
    task.work()
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules_forge\main_thread.py", line 26, in work
    self.result = self.func(*self.args, **self.kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\txt2img.py", line 111, in txt2img_function
    processed = processing.process_images(p)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\processing.py", line 750, in process_images
    res = process_images_inner(p)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\processing.py", line 921, in process_images_inner
    samples_ddim = p.sample(conditioning=p.c, unconditional_conditioning=p.uc, seeds=p.seeds, subseeds=p.subseeds, subseed_strength=p.subseed_strength, prompts=p.prompts)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\processing.py", line 1276, in sample
    samples = self.sampler.sample(self, x, conditioning, unconditional_conditioning, image_conditioning=self.txt2img_image_conditioning(x))
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\sd_samplers_kdiffusion.py", line 251, in sample
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\sd_samplers_common.py", line 263, in launch_sampling
    return func()
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\sd_samplers_kdiffusion.py", line 251, in <lambda>
    samples = self.launch_sampling(steps, lambda: self.func(self.model_wrap_cfg, x, extra_args=self.sampler_extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\utils\_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\repositories\k-diffusion\k_diffusion\sampling.py", line 594, in sample_dpmpp_2m
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\sd_samplers_cfg_denoiser.py", line 182, in forward
    denoised = forge_sampler.forge_sample(self, denoiser_params=denoiser_params,
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules_forge\forge_sampler.py", line 82, in forge_sample
    denoised = sampling_function(model, x, timestep, uncond, cond, cond_scale, model_options, seed)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\ldm_patched\modules\samplers.py", line 289, in sampling_function
    cond_pred, uncond_pred = calc_cond_uncond_batch(model, cond, uncond_, x, timestep, model_options)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\ldm_patched\modules\samplers.py", line 256, in calc_cond_uncond_batch
    output = model_options['model_function_wrapper'](model.apply_model, {"input": input_x, "timestep": timestep_, "c": c, "cond_or_uncond": cond_or_uncond}).chunk(batch_chunks)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\scripts\animatediff_infv2v.py", line 148, in mm_sd_forward
    out = apply_model(info["input"][_context], info["timestep"][_context], **info_c)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\ldm_patched\modules\model_base.py", line 89, in apply_model
    model_output = self.diffusion_model(xc, t, context=context, control=control, transformer_options=transformer_options, **extra_conds).float()
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\ldm_patched\ldm\modules\diffusionmodules\openaimodel.py", line 867, in forward
    h = forward_timestep_embed(module, h, emb, context, transformer_options, time_context=time_context, num_video_frames=num_video_frames, image_only_indicator=image_only_indicator)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\ldm_patched\ldm\modules\diffusionmodules\openaimodel.py", line 64, in forward_timestep_embed
    x = modifier(x, 'after', layer, layer_index, ts, transformer_options)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\scripts\animatediff_mm.py", line 85, in mm_block_modifier
    return self.mm.down_blocks[mm_idx0].motion_modules[mm_idx1](x)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\motion_module.py", line 127, in forward
    return self.temporal_transformer(x)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\motion_module.py", line 185, in forward
    hidden_states = block(hidden_states)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\motion_module.py", line 239, in forward
    hidden_states = attention_block(norm_hidden_states) + hidden_states
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\motion_module.py", line 329, in forward
    x = self.pos_encoder(x)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1518, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\venv\lib\site-packages\torch\nn\modules\module.py", line 1527, in _call_impl
    return forward_call(*args, **kwargs)
  File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\extensions\sd-forge-animatediff\motion_module.py", line 264, in forward
    x = x + self.pe[:, :x.size(1)].type(x.dtype)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!
*** Error completing request
*** Arguments: ('task(c7is4p6v5964tqc)', <gradio.routes.Request object at 0x000001E515B6B250>, '((close-up portrait of geraldine viswanathan woman)), absurdres, 8k uhd, ((cherry blossoms bloom, Softly falling to the ground)),  <lora:laksheya-geraldine_viswanathan:0.7>, <lora:detailed_eye:0.25>,  <lora:polyhedron_skinny_all:0.25>', '(((teeth))), 3d, cartoon, anime, sketches, (worst quality, bad quality, child, cropped:1.4) ((monochrome)), ((grayscale)), (bad-hands-5:1.0), (badhandv4:1.0), (easynegative:0.8), (bad-artist-anime:0.8), (bad-artist:0.8), (bad_prompt:0.8), (bad-picture-chill-75v:0.8), (bad_prompt_version2:0.8), (bad_quality:0.8)', [], 30, 'DPM++ 2M Karras', 1, 1, 8.5, 768, 768, False, 0.7, 2, 'Latent', 0, 0, 0, 'Use same checkpoint', 'Use same sampler', '', '', [], 0, False, '', 0.8, -1, False, -1, 0, 0, 0, ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], generated_image=None, mask_image=None, hr_option='Both', enabled=False, module='None', model='None', weight=1, image=None, resize_mode='Crop and Resize', processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode='Balanced', save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], generated_image=None, mask_image=None, hr_option='Both', enabled=False, module='None', model='None', weight=1, image=None, resize_mode='Crop and Resize', processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode='Balanced', save_detected_map=True), ControlNetUnit(input_mode=<InputMode.SIMPLE: 'simple'>, use_preview_as_input=False, batch_image_dir='', batch_mask_dir='', batch_input_gallery=[], batch_mask_gallery=[], generated_image=None, mask_image=None, hr_option='Both', enabled=False, module='None', model='None', weight=1, image=None, resize_mode='Crop and Resize', processor_res=-1, threshold_a=-1, threshold_b=-1, guidance_start=0, guidance_end=1, pixel_perfect=False, control_mode='Balanced', save_detected_map=True), False, 7, 1, 'Constant', 0, 'Constant', 0, 1, 'enable', 'MEAN', 'AD', 1, False, 1.01, 1.02, 0.99, 0.95, False, 256, 2, 0, False, False, 3, 2, 0, 0.35, True, 'bicubic', 'bicubic', False, 0, 'anisotropic', 0, 'reinhard', 100, 0, 'subtract', 0, 0, 'gaussian', 'add', 0, 100, 127, 0, 'hard_clamp', 5, 0, 'None', 'None', False, 'MultiDiffusion', 768, 768, 64, 4, False, 0.5, 2, False, True, False, 1, False, False, False, 1.1, 1.5, 100, 0.7, False, False, True, False, False, 0, 'Gustavosta/MagicPrompt-Stable-Diffusion', '', <scripts.animatediff_ui.AnimateDiffProcess object at 0x000001E515B69E40>, None, False, '0', '0', 'inswapper_128.onnx', 'CodeFormer', 1, True, 'None', 1, 1, False, True, 1, 0, 0, False, 0.5, True, False, 'CUDA', False, 0, 'None', '', None, False, False, 0.5, 0, False, False, 'positive', 'comma', 0, False, False, 'start', '', 1, '', [], 0, '', [], 0, '', [], True, False, False, False, False, False, False, 0, False) {}
    Traceback (most recent call last):
      File "D:\DEV\AI-PROJECTS\stable-diffusion-webui-forge\modules\call_queue.py", line 57, in f
        res = list(func(*args, **kwargs))
    TypeError: 'NoneType' object is not iterable

---
```

**Comment by navin-hariharan** [comment]
@continue-revolution Any updates on the same? Is it a issue from my side?

**Comment by DA-Charlie** [comment]
inside the "motion_module.py" file there is the class "PositionalEncoding", this class has a function called "forward"
before this line "x = x + self.pe[:, :x.size(1)].type(x.dtype)" there should be "self.pe = self.pe.to(x.device)"
so the tensors use the same device, it's working for me now
the function should look like

`def forward(self, x):`
`    self.pe = self.pe.to(x.device)`
`    x = x + self.pe[:, :x.size(1)].type(x.dtype)`
`    return` self.dropout(x)`

Beware of indentations if you copy this.

## MY LAPTOP

Total VRAM 4096 MB, total RAM 40628 MB
Device: cuda:0 NVIDIA GeForce RTX 3050 Ti Laptop GPU : native

**Comment by continue-revolution** [comment]
This issue will be closed and tracked at #7 
I will try to fix this today
